### PR TITLE
Add isArray in devTools utils

### PR DESCRIPTION
**Pull request by behnammodi**
Also here I replaced `Array.isArray` with `isArray` from 'shared' package

### CODE REVIEW

**Comment by bvaughn** [comment]
When Sebastian landed this in #21188, he confirmed that the Closure compiler inlines the wrapper function. Unfortunately DevTools doesn't use the same compilation process, and as a result, the wrapper/indirection doesn't get inlined.

Maybe longer term, we should look into running Closure compiler over DevTools builds as well– but in the short term, I'm not sure the added benefits from this change are worth the extra wrapper.

Maybe we could use a similar approach as `shared/hasOwnProperty` for DevTools (e.g. `react-devtools-shared/src/isArray`) and just export the function directly?

**Comment by behnammodi** [comment]
Yes, I agree, I will do it

**Comment by behnammodi** [comment]
@bvaughn Done

File: packages/react-devtools-shared/src/isArray.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+declare function isArray(a: mixed): boolean %checks(Array.isArray(a));
+
+const isArrayImpl = Array.isArray;
+
+// eslint-disable-next-line no-redeclare
+function isArray(a: mixed): boolean {
+  return isArrayImpl(a);
+}
+
+export default isArray;
```
Comment (bvaughn) [inline packages/react-devtools-shared/src/isArray.js:19]: Oops, sorry. The main reason for suggesting this here: https://github.com/facebook/react/pull/22438#issuecomment-927843218

Was that DevTools doesn't use Closure compiler, so the wrapper function _doesn't get inlined_. For this reason, I was suggesting we should just export/use `isArrayImpl` directly (like we do with `hasOwnProperty`.

Comment (behnammodi) [reply packages/react-devtools-shared/src/isArray.js:19]: I got it

Comment (behnammodi) [reply packages/react-devtools-shared/src/isArray.js:19]: Done

**Review by bvaughn — APPROVED** [review]
👍🏼 Thanks

**Comment by bvaughn** [comment]
I'm not sure why CI isn't properly running on your PR. Each time you push, it looks like only the CLA check and Code Sandbox run.

**Comment by behnammodi** [comment]
> I'm not sure why CI isn't properly running on your PR. Each time you push, it looks like only the CLA check and Code Sandbox run.

Yes, this is exactly my question, maybe CI knows me :)

**Review by bvaughn — APPROVED** [review]
It's fine. This change seems safe, and I ran the tests locally on your branch. Not sure why CI is not cooperating.

**Pull request merged**